pallets/system/src/header_builder.rs

```diff
@@ -125,24 +125,25 @@ pub trait HostedHeaderBuilder {
 			(kate_commitment, block_dims, data_index)
 		};
 
-		let extrinsics: Vec<Vec<u8>> = app_extrinsics.into_iter().map(|e| e.data).collect();
-		let avail_extrinsics = extrinsics
-			.iter()
-			.filter_map(|e| <AvailExtrinsic>::decode(&mut &e[..]).ok())
-			.collect::<Vec<_>>();
-
-		let data_root: [u8; 32] = if avail_extrinsics.len() > 0 {
-			log::debug!("Decoded some avail extrinsics.");
-			let leaves: Vec<[u8; 32]> = avail_extrinsics
-				.iter()
-				.map(|x| Sha256::hash(&x.data))
-				.collect();
-
-			let data_tree = MerkleTree::<Sha256>::from_leaves(&leaves);
-			data_tree.root().expect("Data Root computation failed")
-		} else {
-			Default::default()
-		};
+		let extrinsics: Vec<Vec<u8>> = app_extrinsics.clone().into_iter().map(|e| e.data).collect();
+		// let avail_extrinsics = extrinsics
+		// 	.iter()
+		// 	.filter_map(|e| <AvailExtrinsic>::decode(&mut &e[..]).ok())
+		// 	.collect::<Vec<_>>();
+
+		// let data_root: [u8; 32] = if avail_extrinsics.len() > 0 {
+		// 	log::debug!("Decoded some avail extrinsics.");
+		// 	let leaves: Vec<[u8; 32]> = avail_extrinsics
+		// 		.iter()
+		// 		.map(|x| Sha256::hash(&x.data))
+		// 		.collect();
+
+		// 	let data_tree = MerkleTree::<Sha256>::from_leaves(&leaves);
+		// 	data_tree.root().expect("Data Root computation failed")
+		// } else {
+		// 	Default::default()
+		// };
+		let data_root = data_root_avail(app_extrinsics);
```

Comment:
nit: can we name the func better? like `build_data_root` or something? `Avail` is implicit so we can drop it from most references.